### PR TITLE
added `Process::getLastOutputTime()` method

### PR DESCRIPTION
**Pull request by connorhu**
| Q             | A
| ------------- | ---
| Branch?       | 4.4 for features
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28026 
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Process/Process.php [hunk 1]
```diff
@@ -954,6 +954,16 @@ public function addErrorOutput(string $line)
         fseek($this->stderr, $this->incrementalErrorOutputOffset);
     }
 
+    /**
+     * Gets the last output time in microseconds
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Process.php]: this is in seconds, isn't it?

File: src/Symfony/Component/Process/Process.php [hunk 2]
```diff
@@ -954,6 +954,16 @@ public function addErrorOutput(string $line)
...
+     *
+     * @return float|null The last output time in microseconds or null if it isn't started
+     */
+    public function getLastOutputTime()
```
Comment (derrabus) [inline src/Symfony/Component/Process/Process.php]:
```suggestion
    public function getLastOutputTime(): ?float
```

**Review by derrabus — CHANGES_REQUESTED** [review]

**Comment by derrabus** [comment]
The PR should target the 4.4 branch.

File: src/Symfony/Component/Process/Process.php (hunk 1 above)
Comment (connorhu) [reply src/Symfony/Component/Process/Process.php]: We store the result of `microtime(true)` in the lastOutputTime field which "returns the current Unix timestamp with microseconds"

https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Process/Process.php#L289

BTW: the documentation of the fields are missing in this file.

Comment (derrabus) [reply src/Symfony/Component/Process/Process.php]: Yes, "with microseconds", but not "in microseconds". The value is stored in seconds, but with microsecond precision. That's why it's a float.

Comment (connorhu) [reply src/Symfony/Component/Process/Process.php]: Oh, I see. My bad.

File: src/Symfony/Component/Process/CHANGELOG.md [hunk 3]
```diff
@@ -12,7 +12,8 @@ CHANGELOG
 4.4.0
 -----
 
-* deprecated `Process::inheritEnvironmentVariables()`: env variables are always inherited.
+ * added `Process::getLastOutputTime()` method
```
Comment (Simperfit) [inline src/Symfony/Component/Process/CHANGELOG.md]: this should be after the deprecated

**Review by Simperfit — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Process/Process.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/Process/Process.php]: "Gets the last output time in seconds" would be enough IMHO as the return value is a `float`.

**Review by fabpot — CHANGES_REQUESTED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @connorhu.

**Comment by OskarStark** [comment]
Why was this PR merged in `master` instead of `4.4`?
The [docs issue](https://github.com/symfony/symfony-docs/issues/12194) also mention `5.0-dev`(which is correct, if `master` for the code is correct)

cc @nicolas-grekas

**Comment by fabpot** [comment]
This was merged in master by mistake. I will cherry-pick the commit in 4.4. Thanks for the head up.

**Comment by OskarStark** [comment]
Thanks, I will update the docs issue accordingly 👍

**Comment by connorhu** [comment]
Thank you @fabpot 

I'll keep an eye out for similar simple issues.

**Comment by fabpot** [comment]
@connorhu If you have access to an Amazon AWS EC2 machine, you might want to have a look a https://github.com/symfony/symfony/issues/33183, I think this is mainly about figuring out how to get the credentials from a local storage (having a look at the PHP SDK might be enough).

**Pull request merged**